docs/adr/0012-versioning.md

```diff
@@ -0,0 +1,46 @@
+# 12) Versioning {#adr_0012}
+
+Date: 2021-02-08
+
+## Status
+
+accepted
+
+## Context
+
+WMF does during their release process create a number of branches which are used during the release process to orchestrate the different tools to produce the correct artifacts and to support the continuous deployment to production. The release branches (ex. REL1_35) are cut for MediaWiki and every extension repository with the purpose of maintaining releases and the weekly deployment branches are (ex. wmf/1.36.0-wmf.13) are used to deployments to the different [groups](https://versions.toolforge.org/). 
+
+Both of these types of branches are maintained by the WMF and will be used by the wikibase release strategy for our major releases and deployment. However we need to decide how to maintain our own intermediate releases that will be published on releases.wikimedia.org, dockerhub and tagged on git.
+
+As there we will still rely on the release branches and deployment branches, we don't see any real benefit by maintaining an additional set of branches for minor releases. These releases could still be maintained through git tagging the WMDE controlled repositories at the point in time a release is produced ([similar to how mediawiki is maintaining their release candidates](https://gerrit.wikimedia.org/g/mediawiki/core/+/refs/tags/1.35.0-rc.0)). 
+
+
+As for the tag itself we also need to decide on version-numbering scheme for differentiating releases where the WMF version number does not change. Seeing that MediaWiki is already using the semantic versioning for incrementing their (maintenance releases) releases in between new release branches being cut it could make sense for wikibase to adopt a similar process as we will still rely on the release branches for future releases.
+
+Example: The patch release following 1.35 (1.35.0) will be known as 1.35.1
+
+However adopting the same semantic versioning for Wikibase also presents some additional risks of adding confusion to what is to be included in a release as our release cycle will not always align perfectly with the one of MediaWiki.
```

Comment:
also here: I don't think that I got the point about the semantic versioning reference, or the point is not accurate.